src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.macios.cs

```diff
@@ -20,7 +20,14 @@ async Task<string> InternalSaveAsync(string initialPath, string fileName, Stream
 	{
 		cancellationToken.ThrowIfCancellationRequested();
 		var fileManager = NSFileManager.DefaultManager;
-		var fileUrl = fileManager.GetTemporaryDirectory().Append($"{Guid.NewGuid()}{GetExtension(fileName)}", false);
+		var tempDirectoryPath = fileManager.GetTemporaryDirectory().Append(Guid.NewGuid().ToString(), true);
+		var isDirectoryCreated = fileManager.CreateDirectory(tempDirectoryPath, true, null, out var error);
+		if (!isDirectoryCreated || error is not null)
+		{
+			throw new Exception(error?.LocalizedDescription ?? "Unable to create temp directory.");
```

Comment:
We don't need the null check here, we do it on the `if`
```suggestion
			throw new Exception(error.LocalizedDescription ?? "Unable to create temp directory.");
```

Reply:
That is not correct. We can not guarantee error is not null because we also check for isDirectoryCreated.

Reply:
Strange, on my end Roslyn says that the error will not be `null`
<img width="1011" alt="Screenshot 2023-03-06 at 11 43 40" src="https://user-images.githubusercontent.com/20712372/223144090-64e8198b-4e76-4d9c-b957-e3e5b003dbd0.png">

So checking for `null` isn't the best way to check if there's an error

Reply:
it can be null in case the operation is successful.